### DISCUSSION TITLE
[Idea] Add unit price to quantity discounts

### DISCUSSION THREAD

**Discussion by tswfi** [PrestaShop/PrestaShop#31948] [Ideas]
### The problem

Storefront nicely shows "You save" and "Unit discount" columns for quantity discounts. But it would be great to also show the effective unit price for the product in the same place.

Using Hummingbird theme:

![image](https://user-images.githubusercontent.com/1352132/228242658-5a2a24b2-a0b0-4325-99c0-64715dfaf464.png)

This could also show the unit price like this:

![image](https://user-images.githubusercontent.com/1352132/228243607-432401eb-3c64-475e-a621-f01efac1b3fb.png)

### The solution

SpecificPriceFormatterCore just needs to give out the unit price from its calculations.

Something like this:

```diff
diff --git a/classes/product/SpecificPriceFormatter.php b/classes/product/SpecificPriceFormatter.php
index 3975b82ff7..0b30582a21 100644
--- a/classes/product/SpecificPriceFormatter.php
+++ b/classes/product/SpecificPriceFormatter.php
@@ -162,6 +162,7 @@ class SpecificPriceFormatterCore
         }
 
         $this->specificPrice['save'] = $priceFormatter->format((($initialPrice * $this->specificPrice['quantity']) - ($discountPrice * $this->specificPrice['quantity'])));
+        $this->specificPrice['discounted_unit_price_formatted'] = $priceFormatter->format($discountPrice);
 
         return $this->specificPrice;
     }
```

### Alternatives

Or SpecificPrice could provide a hook so that module developers can easily extend it.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by tswfi** [comment]
And of course designers will probably remove some of the columns depending on the customer needs. 

The main point is that it is close to impossible to get the discounted unit price currently without overriding SpecificPriceFormatter :)

**Reply by MatShir** [reply]
Somehow the total is missing to compare it to the discount total column, right?

**Reply by tswfi** [reply]
Yeah, actually I think the data given to the template should also contain the `initialPrice` which might allow for even more interesting visualizations for the data. (maybe something like "price per unit 12$ was -$13- you save $1") eh, this doesn't understand overline, but anyways I think you get the idea.

Adding more data available for designers would allow them to do great things without requiring changes to core (even from modules)

**Comment by kpodemski** [comment]
I like it. I had to implement it in some of the client's stores but never considered putting it to the core.

@Hlavtox wdyt?

**Comment by MatShir** [comment]
I like the idea about the hook a bit more

**Reply by kpodemski** [reply]
@MatShir I think we should have a hook and extra data by default. It doesn't change anything by default, but it gives theme developers more options :)

**Reply by MatShir** [reply]
Sounds good to me :)

**Comment by tswfi** [comment]
So I've read through this: https://build.prestashop-project.org/news/2023/github-discussions-are-now-enabled-in-the-prestashop-project/  a few times and I still don't know what is the next step.

Should this discussion be converted to a issue? 

Or do I make the pull request and reference this discussion? (and I would hope that the "more data" and "hook" can be split to two different pr:s)

**Reply by kpodemski** [reply]
You can create a PR(s) and refer to the discussion 👍🏻

**Comment by tswfi** [comment]
hmm, mentioning a discussion in the pr does not autolink like in issues? thats not nice..

https://github.com/PrestaShop/PrestaShop/pull/31961

**Reply by MatShir** [reply]
It is a bit new of us, so the process is not entirely clear. But we have two options:
- Linking the discussion to the PR
- Or converting the discussion into an Issue

But both options have about the same result, the issue format on feature doesn't add much value than a discussion

**Reply by tswfi** [reply]
its nice that a PR can mark issue as resolved. Now marking this discussion as "resolved and closed and will be added in version x.x.x" is a manual operation.. or maybe prestonbot could do some automagion :D

**Reply by MatShir** [reply]
True, let's convert the discussion into an issue and see how the workflow happens

**Reply by MatShir** [reply]
You don't have all the thread after a conversion, the new issue has recap well all the discussion/specs
